numpy/ma/core.py

```diff
@@ -371,24 +371,24 @@ def maximum_fill_value(obj):
         raise TypeError(errmsg)
 
 
-def _recursive_set_default_fill_value(dtypedescr):
+def _recursive_set_default_fill_value(dtype):
     deflist = []
-    for currentdescr in dtypedescr:
-        currenttype = currentdescr[1]
-        if isinstance(currenttype, list):
+    for name in dtype.names:
+        currenttype = dtype[name].base
```

Comment:
I am curious how you thought to use `.base` here, rather than the dtype directly.  I can imagine why it might be a good idea, just wondering if you had explicit rationale for it.